akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala

```diff
@@ -785,6 +795,13 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
       expectMsg("x-b-2")
       expectNoMsg(100.millis)
     }
+    "support calling persistAll with Nil" in {
+      val persistentActor = namedPersistentActor[PersistAllNilPersistentActor]
+      persistentActor ! Cmd("x")
+      expectMsg("before-nil")
+      expectMsg("after-nil")
+      expectMsg("x")
```

Comment:
Hm, I wonder if it is confusing... I may be biased as I implemented the current semantics.
`deferAsync` defers in respect to `persist*` calls, not "in general".
If you think we should address this let's open a ticket.